docs/proposals/202105256-cluster-class-and-managed-topologies.md

```diff
@@ -285,21 +285,26 @@ type LocalObjectTemplate struct {
 #### Validations
 ##### ClusterClass
 - For object creation:
+  - (defaulting) if namespace field is empty for a reference, default it to `metadata.Namespace`
+  - all the reference must be in the same namespace of `metadata.Namespace`  
   - `spec.worker.deployments[i].class` field must be unique within a ClusterClass.
 - For object updates:
+  - all the reference must be in the same namespace of `metadata.Namespace`
+  - `spec.worker.deployments[i].class` field must be unique within a ClusterClass.
   - `spec.worker.deployments` supports adding new deployment classes.
 
 ##### Cluster
 - For object creation:
+  - `spec.topology` and `spec.infrastructureRef` cannot be simultaneously set.
   - `spec.topology` and `spec.controlPlaneRef` cannot be simultaneously set.
-  - If `spec.topology.class` is set, `spec.topology.version` cannot be empty and must be a valid semver.
-
+  - If `spec.topology` is set, `spec.topology.class` cannot be empty.
+  - If `spec.topology` is set, `spec.topology.version` cannot be empty and must be a valid semver.  
+  - `spec.topology.worker.machineDeployments[i].name` field must be unique within a Cluster
+  
 - For object updates:
-  - If spec.topology.class is in use, it cannot be unset or modified.
-  - spec.topology.version cannot be unset and must be a valid semver, if being updated.
+  - If `spec.topology.class` is set, it cannot be unset or modified.
+  - `spec.topology.version` cannot be unset and must be a valid semver, if being updated.
   - `spec.topology.version` cannot be downgraded.
-  - `spec.topology.controlPlane.replicas` can be updated.
-  - `spec.topology.worker.machineDeployments[i].replicas` can be updated.
```

Comment:
Is it being removed since it is obvious or is there some other reason for this?

Reply:
I'm removing it because fields are mutable by default.
Also we are usually documenting only what we prevent to change.